File: topic_4/Haskell/10-uniao.hs
-- Recebe duas listas e retorna a união entre elas
elemento x [] = False -- retorna se um elemento está na lista
elemento x (y:ys) 
    | x == y = True
    | otherwise = elemento x ys

uniao [] [] = []
uniao [] xs = xs
uniao xs [] = xs
-- uniao em ordem
--uniao (x:xs) (y:ys)
--    | x < y = x : uniao xs (y:ys)
--    | x > y = y : uniao (x:xs) ys
--    | otherwise = x : uniao xs ys
--
-- uniao sem ser em ordem
uniao xs ys = xs ++ [y | y <- ys, not (elemento y xs)]

main = do
    a <- readLn :: IO [Int]
    b <- readLn :: IO [Int]
    print $ uniao a b

File: topic_4/Python/10-uniao.py
# Recebe duas listas e retorna a união entre elas

def uniao(xs, ys):
    lista = []
    for x in xs:
        lista.append(x)
    for y in ys:
        if y not in lista:
            lista.append(y)
    return lista

# def uniao(xs, ys):
#    lista = []
#    lista = xs.union(ys)

print(uniao ([4,5], [1]) == [4,5,1]) #True
print(uniao ([4,5], [4,2,5]) == [4,5,2]) #True
print(uniao ([1,2,3], [2,4,6]) == [1,2,3,4,6]) #True